### PR TITLE
[ADD] account_management

### PR DESCRIPTION
**Pull request by tivisse**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_portal_sale/models/sale_order.py [hunk 1]
```diff
@@ -9,11 +9,16 @@ class SaleOrder(models.Model):
...
 
     website_url = fields.Char('Website URL', compute='_website_url', help='The full URL to access the document through the website.')
+    payment_transaction_count = fields.Integer(compute='_compute_payment_transaction_count', string="Number of payment transactions")
 
     def _website_url(self):
         for so in self:
             so.website_url = '/my/orders/%s' % (so.id)
 
+    def _compute_payment_transaction_count(self):
+        for order in self:
```
Comment (tde-banana-odoo) [inline addons/website_portal_sale/models/sale_order.py]: Use a read group, even if normally we use it only on a form view, do not forget that strange people code in odoo.

File: addons/payment_ogone/data/ogone.py [hunk 2]
```diff
@@ -67,11 +67,11 @@
     '30001010': "A technical problem occurred, please contact helpdesk",
     '30001011': "A technical problem occurred, please contact helpdesk",
     '30001012': "Card black listed - Contact acquirer",
-    '30001015': "Your merchant's acquirer is temporarily unavailable, please try later or choose another payment method.",
+    '30001015': "Your merchant's acquirer is temporarily unavailable, please try later or choose another payment_acquirer.",
```
Comment (tde-banana-odoo) [inline addons/payment_ogone/data/ogone.py]: Revert as this is specific ogone data.

Lol.

@xmo-odoo

File: addons/website_sale/views/sale_order_views.xml [hunk 3]
```diff
@@ -13,18 +13,6 @@
         </field>
     </record>
 
-    <record id="view_order_form_acquirer" model="ir.ui.view">
```
Comment (tde-banana-odoo) [inline addons/website_sale/views/sale_order_views.xml]: Why this change ? Rebase issue ?

File: addons/website_portal_sale/models/sale_order.py [hunk 4]
```diff
@@ -85,6 +87,17 @@ def action_view_transaction(self):
             })
         return action
 
+    def action_mark_as_paid(self):
+        # Mark directly a sales order as paid if:
```
Comment (tde-banana-odoo) [inline addons/website_portal_sale/models/sale_order.py]: Make a true docstring instead of degueu comment.

File: addons/website_portal_sale/views/sale_order_views.xml [hunk 5]
```diff
@@ -21,6 +21,10 @@
                     <field name="payment_transaction_count" widget="statinfo" string="Transactions"/>
                 </button>
             </xpath>
+            <xpath expr="//button[@name='action_quotation_send']" position="after">
+                <button name="action_mark_as_paid" type="object" string="Mark as Paid" states="sent,sale" class="btn-primary"/>
+                    <!-- attrs="{'invisible': [('payment_acquirer_id.provider', 'in', ['manual', 'transfer'])]}" -->
```
Comment (tde-banana-odoo) [inline addons/website_portal_sale/views/sale_order_views.xml]: Zboob.

File: addons/website_sale/views/report_shop_saleorder.xml [hunk 6]
```diff
@@ -15,10 +15,10 @@
                             <td>
                                 <strong>Payment Acquirer:</strong> <t t-esc="doc.payment_acquirer_id.name"/><br />
                                 <strong>Payment Status:</strong>
-                                <t t-if="doc.payment_acquirer_id.auto_confirm == 'none'">
+                                <t t-if="not doc.payment_acquirer_id.journal_id">
```
Comment (tde-banana-odoo) [inline addons/website_sale/views/report_shop_saleorder.xml]: Probably to move in the "I remove auto confirm" commit.

Comment (tde-banana-odoo) [inline addons/website_sale/views/report_shop_saleorder.xml]: Actually to remove as custom / transfer payments should be pending. Below code should handle it correctly. I think this t-if can be removed and the one just below does not need to check journal_id.

File: addons/website_sale/models/sale_order.py [hunk 7]
```diff
@@ -166,6 +176,19 @@ def _cart_accessories(self):
             accessory_products -= order.website_order_line.mapped('product_id')
             return random.sample(accessory_products, len(accessory_products))
 
+    def action_mark_as_paid(self):
```
Comment (tde-banana-odoo) [inline addons/website_sale/models/sale_order.py]: Pay attention to code previously added in website_portal_sale then re-moved here: just move it directly to the right place to avoid diff.

File: addons/website_sale/models/sale_order.py [hunk 8]
```diff
@@ -23,6 +23,16 @@ class SaleOrder(models.Model):
     payment_acquirer_id = fields.Many2one('payment.acquirer', string='Payment Acquirer', copy=False)
     payment_tx_id = fields.Many2one('payment.transaction', string='Last Transaction', copy=False)
     only_services = fields.Boolean(compute='_compute_cart_info', string='Only Services')
+    can_directly_mark_as_paid = fields.Boolean(compute='_compute_can_directly_mark_as_paid',
```
Comment (tde-banana-odoo) [inline addons/website_sale/models/sale_order.py]: Probably merge https://github.com/odoo/odoo/pull/17067/commits/aec41a40d3b5d33b2639d6bba554fe1f807e59f5 in this one.

File: addons/payment/models/payment_acquirer.py [hunk 9]
```diff
@@ -185,13 +185,22 @@ def _get_feature_support(self):
...
+        vals = self._check_journal_id(vals)
         return super(PaymentAcquirer, self).create(vals)
 
     @api.multi
     def write(self, vals):
         image_resize_images(vals)
+        vals = self._check_journal_id(vals)
         return super(PaymentAcquirer, self).write(vals)
 
+    def _check_journal_id(self, vals):
```
Comment (tde-banana-odoo) [inline addons/payment/models/payment_acquirer.py]: Rebase issue I suppose (to move into its own commit).

File: addons/sale/views/sale_views.xml [hunk 10]
```diff
@@ -350,7 +350,7 @@
                                     <field name="date_order" attrs="{'invisible': [('state', 'in', ['sale', 'done', 'cancel'])]}"/>
                                     <field name="fiscal_position_id" options="{'no_create': True}"/>
                                     <field name="invoice_status"
-                                        attrs="{'invisible': [('state', 'not in', ('sale','done'))]}"/>
+                                        attrs="{'invisible': [('state', 'not in', ('sale','done'))]}" invisible="1"/>
```
Comment (tde-banana-odoo) [inline addons/sale/views/sale_views.xml]: invisible [1=1] here
invisible [('state', 'not in', ('sale','done'))] when activating invoicing

Or try with a states=.., like you want.

File: addons/website_sale/models/sale_order.py [hunk 11]
```diff
@@ -32,7 +32,7 @@ class SaleOrder(models.Model):
                  'payment_acquirer_id', 'payment_acquirer_id.provider')
     def _compute_can_directly_mark_as_paid(self):
         for order in self:
-            order.can_directly_mark_as_paid = order.state in ['sent', 'sale'] and order.payment_tx_id.state != 'done' and order.payment_acquirer_id.provider in ['transfer', 'manual']
+            order.can_directly_mark_as_paid = order.state in ['sent', 'sale'] and order.payment_tx_id and order.payment_acquirer_id.provider in ['transfer', 'manual']
```
Comment (tde-banana-odoo) [inline addons/website_sale/models/sale_order.py]: To put directly in https://github.com/odoo/odoo/pull/17067/commits/83280a288e799aad565849c652b924cf821cbc88to have only one version of the method.

File: addons/payment/models/payment_acquirer.py (hunk 9 above)
Comment (odony) [reply addons/payment/models/payment_acquirer.py]: Hmm, what about the fact that it makes no sense for write(), as we're now resetting the journal every time you edit an acquirer?
Also:
`vals['journal_id'] = default_journal.id`
;-)
